### PR TITLE
Add manifestv2 doc + autogenerated app resource doc

### PR DESCRIPTION
**Pull request by alexAubin**

### CODE REVIEW

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+## Packaging "v2"
+
+
+Packaging v2 is a major improvement for the packaging introduced in YunoHost 11.1
+
+### Motivation 
+
+The motivations for this new format is:
+- to get rid of some boring or unecessarily complex or duplicateds stuff such as saving/loading settings, creating a system user, etc.. during the install/upgrade/restore script, which are common things accross many apps.
+- to get rid of funky historical, not-super-semantic stuff such as `__FINALPATH__` being replaced by `$final_path` (with underscore), or `__PATH__` being replaced by `$path_url`, etc...
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:20]:
```suggestion
- to get rid of funky historical, not-super-semantic stuff such as `__FINALPATH__` being replaced by `$final_path` (with underscore), or `__PATH__` being replaced by `$path_url`, etc.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- formalize some aspects such as the existence of the install dir, data dir, apt dependencies, database, etc ... which in turn unlock possible future developments such as computing the space used by an app at one given moment, auto-upgrading apt dependencies during Debian major migration, etc...
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:21]:
```suggestion
- formalize some aspects such as the existence of the install dir, data dir, apt dependencies, database, etc. which in turn unlock possible future developments such as computing the space used by an app at one given moment, auto-upgrading apt dependencies during Debian major migration, etc.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+Usage: `python3 convert_app_to_packaging_v2.py /path/yo/your/app`
+
+This will edit the file in place and you should then carefully review and iterate over what the script did for you.
+
+
+### Highlights of the new format
+
+- The manifest is now written as toml (`manifest.toml`) instead of json (`manifest.json`). The structure of the manifest itself has been reworked and now include:
+   - The `upstream` section (now mandatory)
+   - An `integration` section meant to formalize what minimum yunohost version is required, the list of supported architectures, declare if SSO/LDAP is supported, typical resource usage. This is meant to be displayed both in the catalog (similar to app stores) and on the app info page after installing the app - though this is still work in progress.
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:42]:
```suggestion
   - An `integration` section meant to formalize what minimum YunoHost version is required, the list of supported architectures, declare if SSO/LDAP is supported, typical resource usage. This is meant to be displayed both in the catalog (similar to app stores) and on the app info page after installing the app - though this is still work in progress. 
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   - An `integration` section meant to formalize what minimum yunohost version is required, the list of supported architectures, declare if SSO/LDAP is supported, typical resource usage. This is meant to be displayed both in the catalog (similar to app stores) and on the app info page after installing the app - though this is still work in progress. 
+   - A `resource` section (detailed below)
+- Install questions are now automatically saved as settings (except user-provided password and other specific infos)
+- All settings are now automatically loaded in app script environments (so e.g. directly define `$domain`, etc.)
+- Auto-enable `set -eu` aka `ynh_abort_if_errors` in appropriate scripts
+- The safety-backup-before-upgrade is now automatically handled by the core
+- Introduce a new `resource` mechanism to formalize some steps and handle them from the core. 
+    - Resources are declared in the `manifest.toml`. 
+    - They are meant to formalize recurring app needs that are to be provisioned/deprovisioned by the core. 
+    - They include for example : system user, apt dependencies, install dir, data dir, port, permissions, SQL database ...
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:51]:
```suggestion
    - They include for example: system user, apt dependencies, install dir, data dir, port, permissions, SQL database... 
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+- Introduce a new `resource` mechanism to formalize some steps and handle them from the core. 
+    - Resources are declared in the `manifest.toml`. 
+    - They are meant to formalize recurring app needs that are to be provisioned/deprovisioned by the core. 
+    - They include for example : system user, apt dependencies, install dir, data dir, port, permissions, SQL database ... 
+    - Each resource is to be provisioned *before* running the install script, deprovisioned *after* the remove script, and automatically upgraded if needed before running the upgrade script (or provisionned if introduced in the new app version, or deprovisioned if removed w.r.t. the previous app version)
+    - More infos about resources and their options and behavior are available in the dedicated doc page
+- Permissions are to be initialized using the answer to the `init_main_permission` install question (replacing the `is_public` question). No need to write a boring `if $is_public ...` to add visitors ! There is a similar mechanism to init other permissions, eg `init_admin_permission` (cf also the doc about the `permission` resource)
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md]:
```suggestion
- Permissions are to be initialized using the answer to the `init_main_permission` install question (replacing the `is_public` question). No need to write a boring `if $is_public...` to add visitors! There is a similar mechanism to init other permissions, e.g. `init_admin_permission` (cf. also the doc about the `permission` resource)
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+- The content of the `doc/` folder is now meant to be integrated in the webadmin (though this is still WIP as of writing this). In particular:
+    - `DESCRIPTION.md` and the `screenshots/` folder are expected to be displayed prior to the app install form (similar to app stores on mobile)
+    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the french version, hence you can create `ADMIN_fr.md`, etc.
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:57]:
```suggestion
    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the French version, hence you can create `ADMIN_fr.md`, etc.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+- The content of the `doc/notification/` folder is meant to replace the `ynh_send_readme_to_admin` mechanics. The integration of these in the webadmin is still work in progress. You can create:
+    - `doc/notifications/pre_install.md` : a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
+    - `doc/notifications/post_install.md` : a note to be displayed after the app install. Typically to explain post-install operations to be performed manually by the admin and that cannot be automated.
+    - `doc/notifications/pre_upgrade.md` : a note to be displayed prior to any upgrade of this app.
+    - `doc/notifications/pre_upgrade.d/{version}.md` : a note to be displayed prior to a specific version upgrade.
+    - `doc/notifications/post_upgrade.md` : a note to be displayed after the app upgrade. For example in the context of Nextcloud, the fact that you may need to re-enable custom modules manually ?
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:63]:
```suggestion
    - `doc/notifications/post_upgrade.md`: a note to be displayed after the app upgrade. For example in the context of Nextcloud, the fact that you may need to re-enable custom modules manually?
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the french version, hence you can create `ADMIN_fr.md`, etc.
+- The content of the `doc/notification/` folder is meant to replace the `ynh_send_readme_to_admin` mechanics. The integration of these in the webadmin is still work in progress. You can create:
+    - `doc/notifications/pre_install.md` : a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
+    - `doc/notifications/post_install.md` : a note to be displayed after the app install. Typically to explain post-install operations to be performed manually by the admin and that cannot be automated.
+    - `doc/notifications/pre_upgrade.md` : a note to be displayed prior to any upgrade of this app.
+    - `doc/notifications/pre_upgrade.d/{version}.md` : a note to be displayed prior to a specific version upgrade.
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:62]:
```suggestion
    - `doc/notifications/pre_upgrade.d/{version}.md`: a note to be displayed prior to a specific version upgrade.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    - `DESCRIPTION.md` and the `screenshots/` folder are expected to be displayed prior to the app install form (similar to app stores on mobile)
+    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the french version, hence you can create `ADMIN_fr.md`, etc.
+- The content of the `doc/notification/` folder is meant to replace the `ynh_send_readme_to_admin` mechanics. The integration of these in the webadmin is still work in progress. You can create:
+    - `doc/notifications/pre_install.md` : a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
+    - `doc/notifications/post_install.md` : a note to be displayed after the app install. Typically to explain post-install operations to be performed manually by the admin and that cannot be automated.
+    - `doc/notifications/pre_upgrade.md` : a note to be displayed prior to any upgrade of this app.
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:61]:
```suggestion
    - `doc/notifications/pre_upgrade.md`: a note to be displayed prior to any upgrade of this app.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    - `DESCRIPTION.md` and the `screenshots/` folder are expected to be displayed prior to the app install form (similar to app stores on mobile)
+    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the french version, hence you can create `ADMIN_fr.md`, etc.
+- The content of the `doc/notification/` folder is meant to replace the `ynh_send_readme_to_admin` mechanics. The integration of these in the webadmin is still work in progress. You can create:
+    - `doc/notifications/pre_install.md` : a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
+    - `doc/notifications/post_install.md` : a note to be displayed after the app install. Typically to explain post-install operations to be performed manually by the admin and that cannot be automated.
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:60]:
```suggestion
    - `doc/notifications/post_install.md`: a note to be displayed after the app install. Typically to explain post-install operations to be performed manually by the admin and that cannot be automated.
```

File: pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+- The content of the `doc/` folder is now meant to be integrated in the webadmin (though this is still WIP as of writing this). In particular:
+    - `DESCRIPTION.md` and the `screenshots/` folder are expected to be displayed prior to the app install form (similar to app stores on mobile)
+    - `ADMIN.md` is expected to be made available somewhere in the info page and should contain technical admin notes. Other pages can be defined (just name it `WHATEVER.md`). Lang codes are also supported following the existing scheme for the README, eg `README_fr.md` is the french version, hence you can create `ADMIN_fr.md`, etc.
+- The content of the `doc/notification/` folder is meant to replace the `ynh_send_readme_to_admin` mechanics. The integration of these in the webadmin is still work in progress. You can create:
+    - `doc/notifications/pre_install.md` : a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
```
Comment (ericgaspar) [inline pages/06.contribute/10.packaging_apps/14.advanced/03.packaging_v2/packaging_v2.md:59]:
```suggestion
    - `doc/notifications/pre_install.md`: a note to be displayed prior to the app install. Typically to warn of something unusual, such as "the app install will automatically add 1GB swap to the system".
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]
Small typos

**Comment by alexAubin** [comment]
Superseded by https://github.com/YunoHost/doc/pull/2162

**Pull request closed without merging**